plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryArrowToPageConverter.java

```diff
@@ -0,0 +1,287 @@
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.arrow.vector.util.TransferPair;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.bigquery.BigQueryUtil.toBigQueryColumnName;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.Decimals.encodeShortScaledValue;
+import static io.trino.spi.type.Decimals.isLongDecimal;
+import static io.trino.spi.type.Decimals.isShortDecimal;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.LongTimestampWithTimeZone.fromEpochMillisAndFraction;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MICROS;
+import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_MILLISECOND;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.Math.floorDiv;
+import static java.lang.Math.floorMod;
+import static java.lang.Math.toIntExact;
+import static java.lang.String.format;
+import static org.apache.arrow.vector.complex.BaseRepeatedValueVector.OFFSET_WIDTH;
+import static org.apache.arrow.vector.types.Types.MinorType.DECIMAL256;
+import static org.apache.arrow.vector.types.Types.MinorType.LIST;
+import static org.apache.arrow.vector.types.Types.MinorType.STRUCT;
+
+public class BigQueryArrowToPageConverter
+        implements AutoCloseable
+{
+    private final VectorSchemaRoot root;
+    private final VectorLoader loader;
+    private final BufferAllocator allocator;
+    private final List<Type> columnTypes;
+    private final List<String> columnNames;
+
+    public BigQueryArrowToPageConverter(Schema schema, BufferAllocator allocator, List<BigQueryColumnHandle> columns)
+    {
+        this.allocator = allocator;
+        this.columnTypes = columns.stream()
+                .map(BigQueryColumnHandle::getTrinoType)
+                .collect(toImmutableList());
+        this.columnNames = columns.stream()
+                .map(BigQueryColumnHandle::getName)
+                .collect(toImmutableList());
+        ImmutableList.Builder<FieldVector> vectorBuilder = ImmutableList.builder();
+        for (Field field : schema.getFields()) {
+            vectorBuilder.add(field.createVector(allocator));
+        }
+        List<FieldVector> vectors = vectorBuilder.build();
```

Comment:
Looks like it's not always like this:
for example we have test `BaseBigQueryConnectorTest.testSkipUnsupportedType()`
And in this case we have 2 columns (from trino) but 3 vectors (from schema)

And I think we kind of safe in this situation because we doesn't rely on ordering or indexes we rely on Field Names:
`root.getVector(toBigQueryColumnName(columnNames.get(column))`

Reply:
Why unsupported columns exist in `schema`?

Reply:
For me it looks like this columns are unsupported in trino but not in BigQuery, so we kind of filter such columns in trino and that's why we have 2 entries in `columns` but from BigQuery perspective we return all 3 `vectors`

I debugged old version avro and behaviour is exactly the same 2 columns but 3 entries in records

Reply:
From BigQueryType.java : 
`// Unsupported BIGNUMERIC types (precision > 38) are filtered in BigQueryClient.getColumns`

Reply:
It's a bug #15093

Reply:
As https://github.com/trinodb/trino/issues/15093 is merged,
verification added:
`verify(vectors.size() == columns.size()`